session_2/cricviz/app/models/cricketer.rb

```diff
@@ -68,14 +81,58 @@ def self.import_classical_batters
   # `ActiveRecord::RecordNotFound` exception with the player's name as
   # the message.
   def self.update_innings(batting_scorecard, bowling_scorecard)
-    raise NotImplementedError
+    batting_scorecard.each do |b|
+      if Cricketer.exists?(name: b[0])
```

Comment:
This function makes two SQL queries for each player - one for `exists?` and one for `find_by`.  The problem with SQL queries is that they take a lot of time, often slowing the website.

Can you write this to make just one query for each player?

There's also a way to write this to make one query _in total_, so try if you can improve it further.

https://www.sitepoint.com/silver-bullet-n1-problem/

Reply:
For making  only one query should I use ``Cricketer.all`` to load the entire db and then search in it for each player.

Reply:
`Cricketer.all` would be one way to go. 

But there can be only 22 players in a match, right? If the database contains thousands of players (as most websites do), loading everyone would be waste of memory.

Hint: The `where` function can take an array as input too. For example:
```ruby
Cricketer.where(county: ['India', 'Australia'])
``` 

Returns players from either India or Australia.